src/include/indexlr-workers.h

```diff
@@ -338,16 +332,22 @@ MinimizeWorker::work()
 
 			auto minimizers = getMinimizers(hashes, w);
 
-			if (stlfr) {
-				std::regex right("^.*#+");
-				std::regex left("/.*");
-
-				read.barcode = std::regex_replace(read.id, right, "");
-				read.barcode = std::regex_replace(read.barcode, left, "");
+			if (read.id.find("BX:Z:") == std::string::npos) {
+				size_t rightPos = read.id.find('#');
+				if (rightPos == std::string::npos) {
+					ss << read.barcode;
+				} else {
+					size_t leftPos = read.id.find('/', rightPos + 1);
+					if (leftPos == std::string::npos) {
+						ss << read.barcode;
+					} else {
+						ss << read.id.substr(rightPos + 1, leftPos - 1 - rightPos);
```

Comment:
Before this line assert
```c++
assert(leftPos < rightPos);
```

Reply:
I removed the assert and used an if statement instead because it should produce `NA  minimizers` instead of failing